gson/src/main/java/com/google/gson/internal/NonNullElementWrapperList.java

```diff
@@ -107,6 +109,16 @@ public int lastIndexOf(Object o) {
     return delegate.lastIndexOf(o);
   }
 
+  @Override
+  public Spliterator<E> spliterator() {
+    return delegate.spliterator();
+  }
```

Comment:
I have also overridden `spliterator` since the `ArrayList` implementation might be more efficient than the default `List` one.

Though I am not completely sure if this is really necessary; it seems the default `List#spliterator` uses a specialized implementation if the list if `RandomAccess` (which is the case here), and that implementation looks quite similar to the `ArrayList` one.

Reply:
Obsolete: Had to remove these overrides again because the minimum Android API level we support does not have them yet.